src/App/App.js

```diff
@@ -82,7 +89,7 @@ class App extends React.Component {
                   <PrivateRoute path='/home' component={Home} authed={this.state.authed} />
                   <PrivateRoute path="/friends" authed={this.state.authed} component={Friends} />
                   <PrivateRoute path="/articles" authed={this.state.authed} component={Articles} />
-                  <PrivateRoute path="/weather" authed={this.state.authed} component={Weather} />
+                  <PrivateRoute path='/weather' component={() => <Weather uid={uid} />} authed={authed} />
```

Comment:
You should be able to just pass `Weather` into your component prop here, given how you have implemented this code